android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/activity/GeckoViewActivityContextDelegate.kt

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.activity
+
+import android.app.Activity
+import android.content.Context
+import mozilla.components.support.base.log.Log
+import org.mozilla.geckoview.GeckoView
+import java.lang.ref.WeakReference
+
+private const val LOGTAG = "GeckoViewActivityContextDelegate"
+
+/**
+ * GeckoViewActivityContextDelegate provides an active Activity to GeckoView or null. Not to be confused
+ * with the runtime delegate of [GeckoActivityDelegate], which is tightly coupled to webauthn.
+ * See bug 1806191 for more information on delegate differences.
+ *
+ * @param contextRef A reference to an active Activity context or null for GeckoView to use.
+ */
+class GeckoViewActivityContextDelegate
+(private val contextRef: WeakReference<Context?>) : GeckoView.ActivityContextDelegate {
```

Comment:
nit: I don't know why, but the linter should have marked this as a warning.

```suggestion
class GeckoViewActivityContextDelegate(
    private val contextRef: WeakReference<Context?>
) : GeckoView.ActivityContextDelegate {
```

Reply:
Thanks! The linter was acting strange on it for me on this line - I think I need to fix my setup.

Reply:
If you try installing [ktlint](https://plugins.jetbrains.com/plugin/15057-ktlint-unofficial-) and [detekt](https://plugins.jetbrains.com/plugin/10761-detekt) in your IDE, that _might_ help?